stats/statsview.cpp

```diff
@@ -747,6 +747,9 @@ void StatsView::RegressionLine::updatePosition()
 		minX = std::max(minX, intersect_x1);
 		maxX = std::min(maxX, intersect_x2);
 	}
+	QPen pen = item->pen();
+	pen.setWidthF(-2.0 * width * (yAxis->toScreen(maxY)-yAxis->toScreen(minY)) / (maxY - minY));
```

Comment:
I'm not very fond of the optics of the screen shot that you posted. I think the line shouldn't cross the axes. Therefore, I suggest doing this with a QPolygonF.

You can clip the polygon with QPolygonF::intersected() function to the plotting area.

That said, is the pen width really the right thing to do here? Note that the width is perpendicular to the line direction, but naively I would assume that the width you want to specify is perpendicular to the y-axis...?